### PR TITLE
[Merged by Bors] - feat(ModelTheory/Order): Golf and implication instances on theories of orders

### PR DESCRIPTION
**Pull request by awainverse**
Shows a series of implications that models of stronger theories of orders are also models of weaker ones
Redefines some theories of orders using `insert`
Proves lemmas for handling theories defined with `insert`, simplifying a few proofs about order theories

---

- [x] depends on: #16338 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/ModelTheory/Complexity.lean [hunk 1]
```diff
@@ -415,19 +415,25 @@ end BoundedFormula
 /-- A theory is universal when it is comprised only of universal sentences - these theories apply
 also to substructures. -/
 class Theory.IsUniversal (T : L.Theory) : Prop where
-  isUniversal_of_mem : ∀ {φ}, φ ∈ T → φ.IsUniversal
+  isUniversal_of_mem : ∀ φ ∈ T, φ.IsUniversal
```
Comment (YaelDillies) [inline Mathlib/ModelTheory/Complexity.lean:418]: Isn't
```suggestion
  isUniversal_of_mem : ∀ ⦃φ⦄, φ ∈ T → φ.IsUniversal
```
the best of both worlds?

File: Mathlib/ModelTheory/Semantics.lean [hunk 2]
```diff
@@ -689,8 +689,13 @@ theorem model_union_iff {T' : L.Theory} : M ⊨ T ∪ T' ↔ M ⊨ T ∧ M ⊨ T
   ⟨fun h => ⟨h.mono Set.subset_union_left, h.mono Set.subset_union_right⟩, fun h =>
     h.1.union h.2⟩
 
+@[simp]
 theorem model_singleton_iff {φ : L.Sentence} : M ⊨ ({φ} : L.Theory) ↔ M ⊨ φ := by simp
 
+@[simp]
+theorem model_insert_iff {φ : L.Sentence} : M ⊨ insert φ T ↔ M ⊨ φ ∧ M ⊨ T := by
```
Comment (YaelDillies) [inline Mathlib/ModelTheory/Semantics.lean:696]: This shouldn't be simp as `M` appears once the LHS and twice on the RHS 
```suggestion
theorem model_insert_iff {φ : L.Sentence} : M ⊨ insert φ T ↔ M ⊨ φ ∧ M ⊨ T := by
```

**Review by YaelDillies — COMMENTED** [review]
Can you please merge master to reduce the diff?

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Comment by awainverse** [comment]
I'm not sure why `merge-conflict` is still there after I've merged master and passed all checks

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**